rfhub2/cli/keywords_importer.py

```diff
@@ -185,67 +203,80 @@ def delete_outdated_collections(
             self.client.delete_collection(collection)
         return collections_to_delete
 
-    def add_collections(self, collections: List[Dict]) -> List[Dict[str, int]]:
+    def add_collections(
+        self, collections: List[CollectionUpdateWithKeywords]
+    ) -> List[Dict[str, int]]:
         """
         Adds collections and keywords from provided list to app.
         :param collections: List of collections object
         :return: list of dictionaries with collection name and number of keywords.
         """
         loaded_collections = []
         for collection in collections:
-            coll_req = self.client.add_collection(collection)
+            coll_req = self.client.add_collection(collection.collection)
             if coll_req[0] != 201:
                 print(coll_req[1]["detail"])
                 raise StopIteration
             collection_id = coll_req[1]["id"]
-            for keyword in collection["keywords"]:
-                keyword["collection_id"] = collection_id
+            for keyword in collection.keywords:
+                keyword = KeywordCreate(
+                    **{**keyword.dict(), "collection_id": collection_id}
+                )
                 self.client.add_keyword(keyword)
             loaded_collections.append(
-                {"name": collection["name"], "keywords": len(collection["keywords"])}
+                {
+                    "name": collection.collection.name,
+                    "keywords": len(collection.keywords),
+                }
             )
             print(
-                f'{collection["name"]} library with {len(collection["keywords"])} keywords loaded.'
+                f"{collection.collection.name} library with {len(collection.keywords)} keywords loaded."
             )
         return loaded_collections
 
-    def _serialise_libdoc(self, libdoc: Dict, path: str, keywords: Dict) -> Dict:
+    def _serialise_libdoc(self, libdoc: Dict, path: str) -> CollectionUpdate:
```

Comment:
I left this one unchanged, but I knwo it's not right. SHould I use `LibraryDocumentation` class as type for `libdoc` instead of Dict?

Reply:
libdoc returned by LibraryDocumentation factory method is instance of libraryDoc class:
`from robot.libdocpkg.model import LibraryDoc`

Reply:
Fixed, thanks for help! :) I think we are ready to to merge it into main statistic branch?